src/parser/expr.rs

```diff
@@ -1,122 +1,284 @@
 use crate::lexer::Token;
 use crate::parser::ast::{Expr, ExprKind, Ident, Lit};
 use crate::parser::error::{ParseError, ParseResult};
-use crate::parser::span::{Spanned, WithSpan};
+use crate::parser::span::{Span, Spanned, WithSpan};
 use crate::parser::Parser;
 use bumpalo::collections::Vec;
+use std::cmp::Ordering;
+
+/// A binary operator.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum BinOp {
+    Add,
+    Sub,
+    Mul,
+    Div,
+
+    IsEQ,
+    IsNE,
+    IsLT,
+    IsLE,
+    IsGT,
+    IsGE,
+
+    Con,
+    Dis,
```

Comment:
We can also make a `BinOp` variant in `ExprKind` I think instead of separate variants.

Reply:
We can bikeshed however, I don't care too much - I picked the names for the 3-letter consistency!  And I agree, it would be nice to move this to the AST module.

Reply:
I really think it should be Or and And, I cannot distinguish these lol